probe-rs/src/semihosting.rs

```diff
@@ -80,8 +80,18 @@ impl GetCommandLineRequest {
         self.0.write(core, &buf)?;
 
         //TODO: abstract this:
-        let reg = core.registers().get_argument_register(0).unwrap();
-        core.write_core_reg(reg.into(), RegisterValue::U32(0u32))?; // signal to target: status = success
+        match core.architecture() {
+            probe_rs_target::Architecture::Xtensa => {
+                core.write_core_reg(crate::RegisterId::from(2), RegisterValue::U32(0u32))?;
```

Comment:
Having to special-case this smells a bit, did I get the register descriptions wrong, or forgot to implement something to make `get_argument_register` working?

Reply:
Agree, will link this comment to the full embedded-test PR so we don't miss it.